github-management/default-branch-migration.md

```diff
@@ -0,0 +1,110 @@
+# Default Branch Migration
+
+This document outlines steps needed to migrate the default branch
+of your repo from `master` to `main`.
+
+Note: This document is currently a work in progress.
+
+If you have questions about the process, reach out to the GitHub Management Team
+on the _#github-management_ channel on slack or open an issue in the [kubernetes/org] repo.
+
+## Prerequisites
+
+- [ ] Ensure that your repo doesn't have too many:
+  - [ ] open PRs. The branch rename will re-trigger prow on _all_ PRs,
```

Comment:
Do we want to specify how many PRs is too many?

Reply:
[Fellow Developer Here]-- To be honest, I'd love to know how many PRs is `too many` 😅

Reply:
It depends on how many jobs the repo has as well though, and how many resources they consume. One very quick unit test job against 20 PRs is vastly cheaper than a single Kubernetes PR test workload.

I think we may want to suggest instead only doing this if the repo has sufficiently low PR volume that they wouldn't mind shutting down job triggering termporarily by disabling the trigger plugin in prow, and that this should be done during the migration.

Reply:
I am going to take a wild guess and say 20+ is too much. Could revise based on data.

We still advise caution or preparing to handle disruption from a burst of CI (either disabling/enabling trigger, or living with a bunch of PR's potentially failing due to capacity exhaustion)

Reply:
Suggested <20 as guidance.